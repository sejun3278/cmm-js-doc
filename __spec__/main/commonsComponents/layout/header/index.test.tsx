import { render } from "@testing-library/react";
import LayoutHeadPage from "src/main/commonsComponents/layout/header";

export default describe("Layout Header Page", () => {
  // 스냅샷
  test("Layout Header Page - Snapshot", () => {
    const { container } = render(<LayoutHeadPage />);
    expect(container).toMatchSnapshot();
  });

  // 모든 태그가 다 존재하는지 검증
  test("Layout Header Page - Snapshot", () => {
    const { container } = render(<LayoutHeadPage />);
    expect(container).toMatchSnapshot();

    // 전체 태그를 감싸는 상위 부모 태그가 있는지 검증
    const wrapperEle = container.querySelector("._layout_header_wrapper_");
    expect(wrapperEle).toBeInTheDocument();

    if (wrapperEle) {
      // link 컴포넌트가 존재하는지 검증
      const linkEle = wrapperEle.querySelector("._layout_header_link_");
      expect(linkEle).toBeInTheDocument(); // 있는지 검증

      if (linkEle) {
        expect(linkEle).toHaveAttribute("href", "/"); // 주소 검증

        // img 태그가 있는지 검증
        const imgEle = linkEle.querySelector("._layout_header_logo_");
        expect(imgEle).toBeInTheDocument();
        expect(imgEle).toHaveAttribute(
          "src",
          "/images/commons/logo/MCM_white_logo.png"
        );
      }
    }
  });
});
