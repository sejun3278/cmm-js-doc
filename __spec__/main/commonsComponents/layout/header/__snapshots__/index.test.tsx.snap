// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Header Page Layout Header Page - Snapshot 1`] = `
<div>
  <header
    class="_layout_header_wrapper_ css-14b3e4t"
  >
    <a
      class="_link_ _layout_header_link_ css-1h5x3dy"
      href="/"
    >
      <img
        class="_image_ _layout_header_logo_ css-msby2z"
        src="/images/commons/logo/MCM_white_logo.png"
      />
    </a>
  </header>
</div>
`;

exports[`Layout Header Page Layout Header Page - Snapshot 2`] = `
<div>
  <header
    class="_layout_header_wrapper_ css-14b3e4t"
  >
    <a
      class="_link_ _layout_header_link_ css-1h5x3dy"
      href="/"
    >
      <img
        class="_image_ _layout_header_logo_ css-msby2z"
        src="/images/commons/logo/MCM_white_logo.png"
      />
    </a>
  </header>
</div>
`;
